### ISSUE TITLE
uTorrent bad request

### ISSUE THREAD

**Issue by Stonedestroyer** [pymedusa/Medusa#2685]
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)

**Branch/Commit:**
Latest stable and latest stable uTorrent
**OS:**
Windows Server 2016
**What you did:**
Added uTorrent and get bad request
**What happened:**
Can't add torrent
**What you expected:**
**Logs:**
In debug logs
```
2017-05-02 03:21:23 WARNING  SNATCHQUEUE-MANUALSNATCH-295685 :: [9c19b10] uTorrent: Unable to send Torrent
2017-05-02 03:21:23 INFO     SNATCHQUEUE-MANUALSNATCH-295685 :: [9c19b10] uTorrent: Bad Request
2017-05-02 03:21:23 INFO     SNATCHQUEUE-MANUALSNATCH-295685 :: [9c19b10] Downloading (redacted) with 10 seeders and 5 leechers and size 42.64 GB from Rarbg
2017-05-02 03:21:23 INFO     SNATCHQUEUE-MANUALSNATCH-295685 :: [9c19b10] Beginning to manual snatch release: (redacted)
```

**Comment by Stonedestroyer** [comment]
Yes the test works correctly. I tried
µTorrent Stable(3.5.0 build 43580). 
I also tried uTorrent 3.4.8 build 42576

**Comment by fernandog** [comment]
No issues here with build 43580

```
2017-05-08 12:32:22 DEBUG    SNATCHQUEUE-MANUALSNATCH-257655 :: [b4a8a54] Calling uTorrent Client
2017-05-08 12:32:43 DEBUG    SNATCHQUEUE-MANUALSNATCH-257655 :: [b4a8a54] uTorrent: Requested a GET connection to http://localhost:9090/gui/ with params: {u'action': u'add-url', u'token': u'DCXzVnSnRsAkXRnr9_x6shcdUOXjWscpFv3jwZa4E4BW36Ym6p1IB4aPEFkAAAAA', u's': u'magnet:?xt=urn:btih:f3dc1d9fdab484587c&dn=Show.S05E20.2016.720p.BrRip.x264--.SiMPLE&tr=udp://coppersurfer.tk:6969/announce&tr=udp://open.demonii.com:1337&tr=udp://exodus.desync.com:6969&tr=udp://9.rarbg.me:2710/announce&tr=udp://glotorrents.pw:6969/announce&tr=udp://tracker.openbittorrent.com:80/announce&tr=udp://9.rarbg.to:2710/announce'} Data: None
2017-05-08 12:32:43 DEBUG    SNATCHQUEUE-MANUALSNATCH-257655 :: [b4a8a54] uTorrent: Response to GET request is {"build":43580}
```

**Comment by fernandog** [comment]
enabled debug log and try again sending new logs

**Comment by Stonedestroyer** [comment]
That is with debug

**Comment by fernandog** [comment]
@Stonedestroyer no. with debug you should see:

`uTorrent: Requested a GET connection to http://localhost:9090/gui/ with params: {u'action': u'add-url'.....`

**Comment by Stonedestroyer** [comment]
`2017-05-09 10:06:04 WARNING  SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] uTorrent: Unable to send Torrent
2017-05-09 10:06:04 INFO     SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] uTorrent: Bad Request
2017-05-09 10:06:04 DEBUG    SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] uTorrent: Requested a GET connection to http://localhost:5003/gui/ with params: {u'action': u'add-url', u'token': u'GP4PufhjTcK6Y6uXWlhI2oTJ4YNo2x_De-yXnA_G1Sfpx4TMhNMLlmt4EVkAAAAA', u's': u'magnet:?xt=urn:btih:1605ca44f02a326b2d950444b2f68e8876005a62&dn=Continuum.S04.1080p.BluRay.x264-YELLOWBiRD%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce&tr=udp://tracker.coppersurfer.tk:6969/announce&tr=udp://tracker.leechers-paradise.org:6969/announce&tr=udp://tracker.zer0day.to:1337/announce&tr=udp://tracker.opentrackr.org:1337/announce&tr=http://tracker.opentrackr.org:1337/announce&tr=udp://p4p.arenabg.com:1337/announce&tr=http://p4p.arenabg.com:1337/announce&tr=udp://explodie.org:6969/announce&tr=udp://9.rarbg.com:2710/announce&tr=http://explodie.org:6969/announce&tr=http://tracker.dler.org:6969/announce&tr=udp://public.popcorn-tracker.org:6969/announce&tr=udp://tracker.internetwarriors.net:1337/announce&tr=udp://ipv4.tracker.harry.lu:80/announce&tr=http://ipv4.tracker.harry.lu:80/announce&tr=udp://mgtracker.org:2710/announce&tr=http://mgtracker.org:6969/announce&tr=udp://tracker.mg64.net:6969/announce&tr=http://tracker.mg64.net:6881/announce&tr=http://torrentsmd.com:8080/announce'} Data: None
2017-05-09 10:06:04 DEBUG    SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] Calling uTorrent Client
2017-05-09 10:06:04 INFO     SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] Downloading Continuum.S04.1080p.BluRay.x264-YELLOWBiRD[rartv] with 5 seeders and 1 leechers and size 19.63 GB from Rarbg
2017-05-09 10:06:04 INFO     SNATCHQUEUE-MANUALSNATCH-258171 :: [8b5eb6f] Beginning to manual snatch release: Continuum.S04.1080p.BluRay.x264-YELLOWBiRD[rartv]`

**Comment by Stonedestroyer** [comment]
Any progress on this issue?

**Comment by medariox** [comment]
@Stonedestroyer 
I've been looking into this and found the issue(s). Will push a fix ASAP.

**Comment by bodskii** [comment]
Sorry if this sound silly, I am new to github. Is this issue fixed and how to I apply this fix. Thanks.

**Comment by p0psicles** [comment]
I do want to mention that we 'fixed' it. But that the cause of it is really a bug in utorrent. Somehow the utorrent webui doesn't accept a magnet link + trackers when the combination exceeds 1024 chars. So now we cut the amount of trackers at 1024. Imo this should be fixed by utorrent.

If you think it's really important to never have to miss out on a tracker or 2, you even might want to consider using a torrent client, which does not have this limitation.

So to summarize. The issue only occurs when sending manget links to the utorrent webui. Where we have made sure the magnet link is always send. And there is only a chance some tracker urls are cut off.

**Comment by fernandog** [comment]
@Stonedestroyer another possible way is to empty setting `tracker list` in Medusa search settings and only use the original trackers

**Comment by bodskii** [comment]
@medariox @p0psicles  Thanks for the quick response. 
I'm so glad I have migrated over to Medusa from Sickrage, really liking the new features.

**Comment by medariox** [comment]
Glad to hear! Closing the issue as it's solved.
